pykafka/simpleconsumer.py

```diff
@@ -206,6 +206,7 @@ def _update(self):
         with self._update_lock:
             self._cluster.update()
             self._setup_partitions_by_leader()
+            self._discover_offset_manager()
```

Comment:
`_update()` also needs to handle updating partition metadata. As it is now, stopping and then restarting a broker causes an infinite loop of failures in `fetch()` because the original `Broker` instance is still held and not connected.